### PR TITLE
Remove attorney capability to change overtime

### LINKED ISSUE #13584: Remove attorney functionality
## User story
User story: As an attorney, I should not have the ability to designate a case as overtime in Caseflow. Only VLJs or SSCs should have the ability to designate or update overtime cases. 

Please remove the "This work product is overtime" checkbox from the submit draft decision for review workflow.

Attorney OT Functionality

## Acceptance criteria
- [ ] Please put this work behind the feature toggle: :overtime_revamp
- [ ] This feature should not be accessible to the following user groups: attorneys
- [ ] Include screenshot(s) in the Github issue if there are front-end changes
- [ ] Update documentation: notify training for documentation updates
- [ ] Attorneys will no longer have the ability to designate OT cases

### Background/context

We are removing this ability for attorneys as we are moving the marking overtime cases functionality to SSCs & VLJs at case assignment. 

Attorneys should no longer be able to designate their own OT cases in Caseflow. 

### Technical notes

Reporting Implications - We are currently only tracking when a case was marked as being worked on OT at the time of the attorney. Any changes to OT-related data will need to be captured. The data will need to be made available for the following Tableau reports: 
The Attorney production and DAS ratings and comments workbooks

Resources/other links
Product Brief: https://docs.google.com/document/d/1OayEvH7KAAojD9XVHkfUHyLdyECC-PkeIsrEx9krqDA/edit#

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves #13584 

### Description
When feature toggle `:overtime_revamp` is enabled, Attorneys will no longer have the ability to designate OT cases.

### Other approach considered
Considered updating the `CaseReviewsController` but `overtime` was well encapsulated within `params`, which is passed along to `AttorneyCaseReview` and `create` https://github.com/department-of-veterans-affairs/caseflow/blob/43562ed9e75b14c6f949802521da9df6a4214c2b/app/models/attorney_case_review.rb#L72
`params.overtime` is never examined, except to check for permitted params https://github.com/department-of-veterans-affairs/caseflow/blob/4b7ca54f05d592ef7d635d86463cefa2f12a92d1/app/controllers/case_reviews_controller.rb#L53
So I decided to keep `overtime` hidden and well-encapsulated within `params`.

The code change was straight-forward in the front-end code: use the current overtime value provided by `appeal.overtime` as the value to pass to the controller.

I considered several places to add tests (such as `spec/feature/queue/overtime_spec.rb`), and ultimately decided to reuse a lot of the setup and test steps provided by an existing test https://github.com/department-of-veterans-affairs/caseflow/blob/c99b229d857d4787b9b7a0f9fe7bfad0dd87ac1c/spec/feature/queue/ama_queue_spec.rb#L408

### Acceptance Criteria
- [x] Code compiles correctly
- [x] This work is behind the feature toggle: :overtime_revamp
- [x] The overtime checkbox should not be accessible to the following user groups: attorneys
- [ ] Update documentation: notify training for documentation updates
- [x] Attorneys will no longer have the ability to designate OT cases

### Testing Plan
0. Enable `overtime_revamp` (`FeatureToggle.enable!(:overtime_revamp)`)
1. Log in as a judge
2. Assign a Legacy case to attorney
   * (try an AMA case)
3. Mark case at Overtime 
   * (try switching the order of 3 and 2)
   * (try not marking the case as overtime)
4. Log in as the attorney to which you assigned the case
5. Go to the Case Details page of that case. Confirm the OT badge exists, if it was marked as OT by the judge (in step 3).
6. In task dropdown, select "Decision ready for review" and go through the process up to the last step (where you fill in the document ID -- see screenshots below). Note that the "This work product is overtime" checkbox is not present.
   * (try disabling `overtime_revamp` (`FeatureToggle.disable!(:overtime_revamp)`), refresh page and go through the process again, and make sure the checkbox appears)
7. Click Continue and a success banner should appear
8. In the Rails console, check the overtime value (`appeal.latest_attorney_case_review.overtime`) should be what the judge marked.

### User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---
image | image

### CODE REVIEW

File: spec/feature/queue/ama_queue_spec.rb [hunk 1]
```diff
@@ -848,6 +852,20 @@ def judge_assign_to_attorney
     end
   end
 
+  it_behaves_like "Judge has a case to assign to an attorney"
```
Comment (yoomlam) [inline spec/feature/queue/ama_queue_spec.rb:855]: Run original test (now converted into a `shared_example`).

File: spec/feature/queue/case_details/update_document_id_spec.rb [hunk 2]
```diff
@@ -147,7 +147,6 @@ def create_legacy_attorney_case_review_in_browser(appeal)
     click_on "#{appeal.veteran_full_name} (#{appeal.sanitized_vbms_id})"
     click_dropdown(index: 1)
     click_label("omo-type_OMO - VHA")
-    click_label("overtime")
```
Comment (yoomlam) [inline spec/feature/queue/case_details/update_document_id_spec.rb:150]: No longer displayed.

File: spec/feature/queue/attorney_checkout_flow_spec.rb [hunk 3]
```diff
@@ -600,7 +600,6 @@
         expect(page).to have_content("Submit OMO for Review")
 
         click_label("omo-type_OMO - VHA")
-        click_label("overtime")
```
[the remark this answers is not in the document]
Comment (hschallhorn) [reply spec/feature/queue/attorney_checkout_flow_spec.rb:603]: Shouldn't these both still be displayed as the feature toggle is not on here?

File: spec/feature/queue/case_details/update_document_id_spec.rb (hunk 2 above)
Comment (hschallhorn) [reply spec/feature/queue/case_details/update_document_id_spec.rb:150]: Shouldn't these both still be displayed as the feature toggle is not on here?

File: spec/feature/queue/attorney_checkout_flow_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (yoomlam) [reply spec/feature/queue/attorney_checkout_flow_spec.rb:603]: You're right! I'll change it to `click_label("overtime") unless FeatureToggle.enabled?(:overtime_revamp)`

[the remark this answers is not in the document]
Comment (hschallhorn) [reply spec/feature/queue/attorney_checkout_flow_spec.rb:603]: I think we should leave out the unless. This ensures that the feature toggle is not enabled when we don't expect it to be (test will fail if it is enabled here)

[the remark this answers is not in the document]
Comment (yoomlam) [reply spec/feature/queue/attorney_checkout_flow_spec.rb:603]: Done and tested.

**Comment by yoomlam** [comment]
For reference: https://github.com/department-of-veterans-affairs/caseflow/issues/13584#issuecomment-626045465
> We still want to record the current overtime status at the time of attorney checkout. `appeal.overtime?` is used for displaying the OT badge and could still potentially be toggled, e.g. subsequent rewrites from quality review or dispatch.

So, keep `AttorneyCaseReview.overtime`.

**Pull request merged**